searches/ternary_search.py

```diff
@@ -69,35 +148,20 @@ def rec_ternary_search(left, right, A, target):
             return rec_ternary_search(left, oneThird - 1, A, target)
         elif A[twoThird] < target:
             return rec_ternary_search(twoThird + 1, right, A, target)
-
         else:
             return rec_ternary_search(oneThird + 1, twoThird - 1, A, target)
     else:
-        return None
-
-
-# This function is to check if the array is sorted.
-def __assert_sorted(collection):
-    if collection != sorted(collection):
-        raise ValueError("Collection must be sorted")
-    return True
+        return -1
 
 
 if __name__ == "__main__":
-    user_input = input("Enter numbers separated by coma:\n").strip()
-    collection = [int(item) for item in user_input.split(",")]
-
-    try:
-        __assert_sorted(collection)
-    except ValueError:
-        sys.exit("Sequence must be sorted to apply the ternary search")
-
-    target_input = input("Enter a single number to be found in the list:\n")
-    target = int(target_input)
+    user_input = input("Enter numbers separated by comma:\n").strip()
+    collection = sorted(int(item.strip()) for item in user_input.split(","))
+    target = int(input("Enter the number to be found in the list:\n").strip())
     result1 = ite_ternary_search(collection, target)
     result2 = rec_ternary_search(0, len(collection) - 1, collection, target)
 
-    if result2 is not None:
+    if result2 != -1:
         print(f"Iterative search: {target} found at positions: {result1}")
         print(f"Recursive search: {target} found at positions: {result2}")
```

Comment:
Does this provide the right answer if the user inputs an out-of-order list?

Reply:
It does not, I think it is better to return to the _assert\_sorted_ logic.

Reply:
OK but let's do it inline.  We don't need a separate function.

Reply:
I guess my error could be avoided by testing that part of the code,  however the use of inputs make it hard to test. Do you know any way that the **main** could be tested ?

Reply:
Sure.
def perform_searches(collection: List[int], target: int) -> None:
    # put lines 159, 160, and 162-168 under this function and then you can test `perform_searches()` all day long.

Reply:
Ok, I guess thats the same than testing each one of the two searches functions.
I meant something to try to test different user inputs via the input function, not when the collection is already defined.